test/measurement/discriminator/test_iq_discriminator.py

```diff
@@ -169,3 +170,87 @@ def test_is_calibration(self):
         self.assertFalse(discriminator.is_calibration('cal01101b'))
         self.assertFalse(discriminator.is_calibration('test'))
         self.assertFalse(discriminator.is_calibration('_cal_2121'))
+
+
+class TestSklearnIQDiscriminator(unittest.TestCase):
+    """
+    Test methods of the sklearn IQ discriminators.
+    """
+
+    def setUp(self):
+        """
+        Setup internal variables and a fake simulation. Aer is used to get the
+        structure of the qiskit.Result. The IQ data is generated using gaussian
+        random number generators.
+        """
+        self.shots = 52
+        self.qubits = [0, 1]
+
+        meas_cal, _ = circuits.tensored_meas_cal([[0], [1]])
+
+        backend = Aer.get_backend('qasm_simulator')
+        job = qiskit.execute(meas_cal, backend=backend, shots=self.shots,
+                             meas_level=1)
+
+        self.cal_results = job.result()
+
+        i0, q0, i1, q1 = 0., -1., 0., 1.
+        ground = utils.create_shots(i0, q0, 0.1, 0.1, self.shots, self.qubits)
+        excited = utils.create_shots(i1, q1, 0.1, 0.1, self.shots, self.qubits)
+
+        self.cal_results.results[0].meas_level = 1
+        self.cal_results.results[1].meas_level = 1
+        self.cal_results.results[0].data = ExperimentResultData(memory=ground)
+        self.cal_results.results[1].data = ExperimentResultData(memory=excited)
```

Comment:
This overlaps a lot with `setUp` from `TestLinearIQDiscriminator`. This would be a good occasion to remove this redundancy by
1. Introducing `BaseTestIQDiscriminator` with a `setUp` method
2. Creating classes that inherit from `BaseTestIQDiscriminator`
    - `class TestLinearIQDiscriminator(BaseTestIQDiscriminator)`
    - `class TestSklearnIQDiscriminator(BaseTestIQDiscriminator)`

Reply:
Done.